### ISSUE TITLE
Widget API - welcome message

### ISSUE THREAD

**Issue by dragosite** [RocketChat/Rocket.Chat#33918]
Hi there,

We are using the Widget API to register a visitor and assign a department and agent. We already know the name and email so we want to automate this which works great. However we want to send an initial message from the Agent, like a welcome message that is automated enough to be able to include name and email.

The triggers functionality doesn't provide this, as well as it doesn't seem to play well with the Widget API either (for example this.maximizeWidget(); doesn't work with the "chat opened by a visitor" in triggers).

To summarise, the idea is to register the visitor programatically and send an automated welcome message from an agent, ie. "Welcome {name}, how can we help?"

Thanks

**Comment by ujjwaldubey1** [comment]
can i work on it?

**Comment by dragosite** [comment]
@ujjwaldubey1 Do you have a solution for this?

**Comment by reetp** [comment]
Plesse note that Github is for issues in Rocket.Chat, not general support.

Please ask at 

https://open.rocket.chat/channel/support

You might want to check the rocket API docs too.

https://developer.rocket.chat

**Comment by reetp** [comment]
> can i work on it?

Kindly refrain from leaping in unless you really know what you are doing.

You need to properly triage whether a bug actually exists, and then check with the team if it is something they are working on themselves or not.

Unlike many projects, github is not the primary issue tracker at Rocket.Chat

Please ask for more information on https://open.rocket.chat/channel/support

**Comment by ujjwaldubey1** [comment]
@dragosite I Think I'll be able to implement this .

**Comment by sasibhumaraju** [comment]
Good day! @dragosite 😊

Could you please provide simulation of this or any screen shots.

**Comment by dragosite** [comment]
Hi @sasibhumaraju 

I've tried using the Widget API to register and open the widget:

```
RocketChat(function() {
    this.registerGuest({
        name: 'Dragos Test',
        email: 'dragos@test.com',
        department: 'department_name',
        customFields: [
            {key:  'question1', value: 'Test123', overwrite: true},
            {key:  'answer1', value: 'test123', overwrite: true}
        ]
    });
    this.setAgent({
        _id: '1hHnqY3E24aYEQSSj',
        username: 'test',
    });
    this.maximizeWidget();
});
```

In the second step, I created a trigger to send a welcome message on the "chat opened by a visitor" event, but it fails because my code above cancels the trigger.

The closest I got was to register the visitor in a previous step and then send the welcome message on a timer on the next page. However, the message looks strange and doesn't open properly as a nice conversation.

Thanks,
Dragos

**Comment by reetp** [comment]
Please note - if you think there is a bug then fill out a bug template with your server information and evidence of errors.

We have absolutely no idea what version you are using or setup so everything here is guesswork.

It appears that this is almost certainly a support issue there is no evidence that anything is broken - you just can't get it to work the way you want it to. Please use the appropriate channels for that as indicated.

https://open.rocket.chat/channel/support

or possibly better here:

https://open.rocket.chat/channel/omnichannel

Thank you.

**Comment by dragosite** [comment]
Hi @reetp 

I am willing to pay for the Pro package to be eligible for support but I still don't know if it can do what I need it for, and secondly, it is forcing me to buy 51 seats as a minimum which is US$2,448.00 per year. :) I only have 3 registered users on my self hosted K8s instance.

```
Deployment
Version
7.0.0
Deployment ID
57a1178d-5c6d-4d1c-b82b-07852e9214e6
Cloud Workspace ID
67164a1e1296318951086d05
Apps Engine Version
1.47.0
Node Version
v20.17.0
Database Migration
308 (November 12, 2024 11:54 AM)
MongoDB
8.0.1 / wiredTiger (oplog Disabled)
Commit Details
HEAD: (dfc7e0e87)
Branch: HEAD
Release 7.0.0
PID
1

```

**Comment by reetp** [comment]
> 8.0.1 / wiredTiger 

So, first Mongo 8 is not officially supported.

https://github.com/RocketChat/Rocket.Chat/releases/tag/7.0.0

> (oplog Disabled)

Next, you should have a replicaset and oplogs on with the correct oplog URL. Might not fix your issue but it is the recommended setup. never go off piste unnecessarily.

https://docs.rocket.chat/docs/configure-a-replica-set-for-mongodb

But as I have said, this is not a support forum. It is a bug tracker.

> I am willing to pay for the Pro package to be eligible for support, however, I still don't know if it can do what I need it for, and secondly, it is forcing me to buy 51 seats as a minimum, which is US$2,448.00 per year. :) I only have 3 registered users on my self-hosted K8s cluster.

Such is the nature of open source.

As I have said more than once, the first place to ask for support is on open.rocket.chat 
That is your first port of call. I have linked the channels already.

Regarding purchasing a subscription I would suggest you speak directly to sales and explain your predicament.

Either way, this is not a bug as far as i can see.

**Comment by dragosite** [comment]
Hi @reetp 

Thank you for your suggestions. Didn't realise the Mongo version bit so thanks.

Unless intentional, the bug here would be that `this.maximizeWidget();` method doesn't trigger the "chat opened by a visitor" Trigger event.

The feature requests are to connect WidgetAPI to Triggers, as well as have a way of impersonating Agents and sending messages via API. I couldn't find that anywhere in the API unless I missed it. The only way to send a message that I could find was via Triggers which is not connected to the API so we can further automate.

The main purpose of my post is that I've seen this or similar questions asked many times with no answers, and hoping a fellow user might see this and point me in the right direction.

Thanks,
Dragos

**Comment by reetp** [comment]
You don't need to `@` anyone - I see all the comments and react at my leisure.

I am not going to answer any further here. Please use open where your question belongs. Thanks.
